src/lib/shared/configmap.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const
+  _     = require("lodash"),
+  file  = require('../shared/file'),
+  utils = require('../shared/utils')
+
+module.exports = class Configmap {
+  constructor(revision, componentName) {
+    this.revision  = revision
+    this.errors    = []
+    this.rationale = []
+    this.valid     = true
+
+    return file.readAt(
```

Comment:
It's a pretty bad anti-pattern to return anything (esp. something that mutates the return value of 'new') from a constructor. If you want to start something when you construct the object, you could do something like:
```
constructor(args) {
    ....
    this.initPromise = doSomething();
}

callAfterConstruction() {
    return this.initPromise;
}
```
That way you cleanly return from the constructor ASAP as intended, but can then call a method that will either have resolved or will return once the task started in the constructor has resolved.